client.go

```diff
@@ -119,12 +158,69 @@ func NewClient(addr string, options ...func(c *Client) error) (*Client, error) {
 		return nil, c.scanErr()
 	}
 
+	if err := c.conn.SetReadDeadline(time.Time{}); err != nil {
+		return nil, err
+	}
+
+	// Start handlers
+	go c.messageHandler()
+	go c.workHandler()
+
 	return c, nil
 }
 
-// setDeadline updates the deadline on the connection based on the clients configured timeout.
-func (c *Client) setDeadline() error {
-	return c.conn.SetDeadline(time.Now().Add(c.timeout))
+// messageHandler scans incoming lines and handles them accordingly.
+func (c *Client) messageHandler() {
+	for {
+		if c.scanner.Scan() {
+			line := c.scanner.Text()
+			if line == "error id=0 msg=ok" {
+				c.err <- nil
+			} else if matches := respTrailerRe.FindStringSubmatch(line); len(matches) == 4 {
+				c.err <- NewError(matches)
+			} else if strings.Index(line, "notify") == 0 {
+				if n, err := decodeNotification(line); err == nil {
+					// non-blocking write
+					select {
+					case c.notify <- n:
+					default:
+					}
+				}
+			} else {
+				c.res = append(c.res, line)
+			}
+		} else {
```

Comment:
Not sure what you mean. Timeout errors? Any errors are returned with `c.err <- err`

Reply:
yes, but as you're always reading, in order to facilitate notifications, and every read should have a timeout (not just the ones triggered by writes) then you need to deal with the expected timeouts otherwise the client would get a constant stream of errors even when they didn't send any commands.